### ISSUE TITLE
Setting transform of meshes in MultiMesh produces an error

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#12595]
Godot 3.0 alpha 2, Windows 10 64 bits
nVidia GeForce 940M

As an attempt to answer @abhaskumarsinha, I found out this code doesn't work:

```gdscript
extends MultiMeshInstance

func _ready():
	# Create multimesh resource that will contain 4 cubes
	var mm = MultiMesh.new()
	mm.instance_count = 4
	mm.transform_format = MultiMesh.TRANSFORM_3D
	mm.mesh = preload("res://cube.obj")
	
	# Set position of all cubes at random
	for i in range(mm.instance_count):
		var pos = Vector3(rand_range(-2, 2), rand_range(-2, 2), rand_range(-2, 2))
		var t = Transform(Basis(), pos)
		mm.set_instance_transform(i, t)
	
	# Assign multimesh to be rendered by the MultiMeshInstance
	self.multimesh = mm
```

It shows only one cube, and 4 errors are logged, everytime instance transforms are set:
```
ERROR: RasterizerStorageGLES3::multimesh_instance_set_transform: Condition ' multimesh->transform_format == VisualServer::MULTIMESH_TRANSFORM_2D ' is true.
   At: drivers\gles3\rasterizer_storage_gles3.cpp:3825
```

But as shown in the code, I did set the format to 3D...

Edit:
It works if I assign `transform_format` before setting `instance_count`. I didn't expect these to be order dependent.

**Comment by djrm** [comment]
it is clear why you need to call transform_format and color_format, before setting the instance count, (it allocates and configures everything when you do that) but the name of the method is not very descriptive, another option would be to configure again when setting the formats.

**Comment by Zylann** [comment]
If there is an order to this, I think exposing the API as properties is a bad idea. Because these are also exposed in the inspector, I wonder which kind of magic Godot does to keep them assigned in order...

**Comment by reduz** [comment]
This is not really a bug, but an error should be shown if format or something is changed before instance count. Is an error being shown or should we add it?

**Comment by Zylann** [comment]
@reduz only this, in the console as usual:
```
ERROR: RasterizerStorageGLES3::multimesh_instance_set_transform: Condition ' multimesh->transform_format == VisualServer::MULTIMESH_TRANSFORM_2D ' is true.
   At: drivers\gles3\rasterizer_storage_gles3.cpp:3825
```
My main surprise was the order of assignment, which is usually not implied with properties in general.

**Comment by akien-mga** [comment]
We have now entered release freeze for Godot 3.0 and want to focus only on release critical issues for that milestone. Therefore, we're moving this issue to the 3.1 milestone, though a fix may be made available for a 3.0.x maintenance release after it has been tested in the master branch during 3.1 development. If you consider that this issue is critical enough to warrant blocking the 3.0 release until fixed, please comment so that we can assess it more in-depth.
